src/smartcontracts/codeMetadata.ts

```diff
@@ -100,13 +119,13 @@ export class CodeMetadata {
     }
 }
 
-enum ByteZero {
+export enum ByteZero {
```

Comment:
If exported, perhaps we should rename them to be more explicit? E.g. `CodeMetadataByteZero` etc.?

Is it required to export them (this is a low-level detail of `CodeMetadata`)? Or perhaps they are only exported for the sake of the tests?

Reply:
I agree with the renaming if they really need to be exported. Maybe some other way of doing was to keep the properties private and have setters and getters, or maybe some methods like `isUpgradeable(): boolean`. Just a thought.

Reply:
Good point, seeing it now I dislike it as well. I will propose a new way by placing Byte info in the CodeMetadata namespace

Reply:
Updated. Byte info is now only exposed through the CodeMetadata namespace, as in: `CodeMetadata.ByteZero.Upgradeable`